### DISCUSSION TITLE
Have an extractVersion like config for changelog version matching only

### DISCUSSION THREAD

**Discussion by viceice** [renovatebot/renovate#24900] [Suggest an Idea]
**What would you like Renovate to be able to do?**

I would like to have a feature like the [`extractVersion`](https://docs.renovatebot.com/configuration-options/#extractversion) but that should only used for changelog retrieval.
With that way we can extract the version from eg `1.2.3-alpine` to `1.2.3` to match against versions in changelogs, so renovate would be able to fetch the corresponding changes for an update like `node:12.8.1-alpine` to `node:12.10.0-alpine`.

**Did you already have any implementation ideas?**
 not yet

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

not known to me.

**Is this a feature you'd be interested in implementing yourself?**

sure

**Comment by viceice** [comment]
example:

![image](https://user-images.githubusercontent.com/1798109/94920726-1ef79380-04b7-11eb-8c26-29d640e0eb06.png)

https://github.com/GoogleContainerTools/kaniko

They have a chgangelog and releases.

Maybe there is a bug and renovate should already handle this.

**Comment by viceice** [comment]
heres a sample pr: https://github.com/viceice-tests/renovate-changelog-kaniko/pull/2

**Comment by adamwilner** [comment]
did you ever figure out how to do this?

**Reply by viceice** [reply]
no
